src/Zynga/Framework/PgData/V1/PgModel/Reader.hh

```diff
@@ -325,13 +332,18 @@ class Reader implements ReaderInterface {
       return null;
 
     } catch (Exception $e) {
```

Comment:
new check added here. there is no harm in calling unlock if it is not locked already by current thread.
As you mentioned above, we don't miss if it is locked & exception occurred